com.unity.render-pipelines.core/Tests/Runtime/CollectionExtensionTests.cs

```diff
@@ -0,0 +1,141 @@
+using System.Collections;
+using NUnit.Framework;
+using System.Collections.Generic;
+
+namespace UnityEngine.Rendering.Tests
+{
+    public class CollectionExtensionTests
+    {
+        static TestCaseData[] s_TestCaseDatas =
+        {
+            new TestCaseData(new List<int>(), 0)
+                .Returns(new List<int>())
+                .SetName("Empty"),
+            new TestCaseData(new List<int>(), -1)
+                .Returns(new List<int>())
+                .SetName("EmptyNegative"),
+            new TestCaseData(new List<int>() {1,2}, -1)
+                .Returns(new List<int>() {1,2})
+                .SetName("NonEmptyNegative"),
+            new TestCaseData(new List<int>() {1,2}, 2)
+                .Returns(new List<int>())
+                .SetName("All"),
+            new TestCaseData(new List<int>() {1,2}, 1)
+                .Returns(new List<int>() {1})
+                .SetName("JustLast"),
+            new TestCaseData(new List<int>() {1,2}, 3)
+                .Returns(new List<int>())
+                .SetName("BiggerThanCollection"),
+            new TestCaseData(new List<int>() {1,2,3,4,5,6,7}, 3)
+                .Returns(new List<int>(){1,2,3,4})
+                .SetName("SomeElements"),
+        };
+
+        [Test, TestCaseSource(nameof(s_TestCaseDatas))]
+        public List<int> TestsList(List<int> ints, int elementsToRemove)
```

Comment:
Can you refactor this as property tests?

like

```csharp
public bool IsRangeRemoved<TList, TValue>(TList list, uint startIndex, uint count) 
    where TList: IList<TValue>
    where TValue: IEquatable<TValue> 
    {
         using (ListPool<TValue>.Get(out var copy)
        {
             copy.AddRange(list);
             var success = list.TryRemoveRange(startIndex, count);
         
             var expectSuccess = startIndex + count < list.Count;
             var isPropertyValid = true;
             if (expectSuccess)
             {
                 // check values here, on fail isPropertyValid = false
             }
             return isPropertyValid;
        {
    }

[Test, TestCaseSource( /*list test case source*/)]
public IsRangedRemovedOnList(List<int> ints, uint start, uint count) => IsRangeRemoved(ints, start, count);
```

Reply:
Use this version to create a single test using simplelist and list<>